cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -4511,6 +4511,7 @@ endif()
 
 macro(build_opentelemetry)
   message(STATUS "Building OpenTelemetry from source")
+  message(WARNING "This can lead to build errors due to conflicting abseil (GH-36013).")
```

How about reporting an error if Protobuf is 3.22 or later?

```suggestion
  if(Protobuf_VERSION VERSION_GREATER_EQUAL 3.22)
    message(FATAL_ERROR "GH-36013: Can't use bundled OpenTelemetry with Protobuf 3.22 or later: ${Protobuf_VERSION}")
  endif()
```

---

That looks good